src/common/aes/aes.c

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <oqs/common.h>
+
+#include "aes.h"
+#include "aes_local.h"
+
+void OQS_AES128_ECB_load_schedule(const uint8_t *key, void **_schedule, UNUSED int for_encryption) {
+	#if defined(OQS_USE_AES_INSTRUCTIONS)
```

Comment:
The logic here should be:

```c
#if defined(OQS_USE_CPU_EXTENSIONS)

#if defined(OQS_PORTABLE_BUILD)
OQS_CPU_EXTENSIONS available_cpu_extensions = OQS_get_available_CPU_extensions();
if (available_cpu_extensions.AES_ENABLED) {
#endif /* OQS_PORTABLE_BUILD */
    oqs_aes128_load_schedule_ni(key, _schedule);
#if defined(OQS_PORTABLE_BUILD)
} else {
    oqs_aes128_load_schedule_c(key, _schedule, for_encryption);
#endif /* OQS_PORTABLE_BUILD */
#else
    oqs_aes128_load_schedule_c(key, _schedule, for_encryption);

#endif /*OQS_USE_CPU_EXTENSIONS */
```

Also, the `OQS_USE_*_INSTRUCTIONS` variables and #define's are indeed confusing. Perhaps it would be better to rename them to `OQS_COMPILE_WITH_*_INSTRUCTIONS`.

Reply:
Thanks.  I've pushed two commits, one that directly applies the logic you gave above, and then one that puts it into a macro: since I know I'm going to need to use the same logic many times in that file, I thought a macro might be cleaner. Can you take a look and see if the macros matches the logic?

Reply:
Agreed, it is cleaner. And it does match the logic.